xla/service/buffer_assignment.cc

```diff
@@ -251,9 +251,9 @@ void BufferAllocation::AddAssignment(const HloValue& buffer, int64_t offset,
   CHECK_LE(offset + size, size_)
       << "LogicalBuffer " << buffer
       << " size out of range at offset: " << offset << " with size: " << size;
-  CHECK_EQ(buffer.color(), color())
-      << "Buffer color " << buffer.color() << " for buffer " << buffer
-      << " does not match allocation color " << color() << ".";
+  // CHECK_EQ(buffer.color(), color())
```

Comment:
Should this be commented out?

Reply:
For temp buffer with new memory space color, it will no longer be valid. I changed the assertion like below: 
  
  if (!(IsPreallocatedTempBuffer() &&
        color() == kPersistentTempMemorySpaceColor)) {
    CHECK_EQ(buffer.color(), color())
        << "Buffer color " << buffer.color() << " for buffer " << buffer
        << " does not match allocation color " << color() << ".";
  }